plugins/inputs/apache/apache.go

```diff
@@ -61,21 +84,47 @@ func (n *Apache) Gather(acc telegraf.Accumulator) error {
 	return outerr
 }
 
-var tr = &http.Transport{
-	ResponseHeaderTimeout: time.Duration(3 * time.Second),
-}
+func (n *Apache) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	var err error
+	var resp *http.Response
+	var req *http.Request
+	var tr *http.Transport
+	var client *http.Client
+
+	if addr.Scheme == "https" {
+		//fmt.Printf("SSL skip verify = true\n")
+		tr = &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+			TLSClientConfig:       &tls.Config{InsecureSkipVerify: n.Sslskipverify},
+		}
+	} else {
+		//fmt.Printf("no verify configured\n")
+		tr = &http.Transport{
+			ResponseHeaderTimeout: time.Duration(n.Timeout) * time.Second,
+		}
+	}
 
-var client = &http.Client{
-	Transport: tr,
-	Timeout:   time.Duration(4 * time.Second),
-}
+	client = &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(n.Timeout) * time.Second,
+	}
 
-func (n *Apache) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
-	resp, err := client.Get(addr.String())
-	if err != nil {
-		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	if n.useHTTPAuth {
```

Comment:
it's not necessary to set a boolean value like this, I think you can change the plugin to create a `req` object in all instances, and then just call `req.SetBasicAuth` only if Username & Password are not equal to `""`